src/app/functionality/run_activy.py

```
@@ -18,13 +18,17 @@ def run_activy():
     try:
-        stage = tryCheckStage(device, ["mainMenu", "login"], templatesGeneralNavigation, retries=100)
+        stage = try_check_stage(device, ["mainMenu", "login"], templatesGeneralNavigation, retries=100)
         if stage == "mainMenu":
             logout(device)
-    except ValueError as e:
+    except TimeoutError as e:
         update_needed = update_app(device)
         if not update_needed:
             raise e
+        else:
+            stage = tryCheckStage(device, ["mainMenu", "login"], templatesGeneralNavigation, retries=100)
```

Ruff (0.11.9)

Undefined name `tryCheckStage`

(F821)